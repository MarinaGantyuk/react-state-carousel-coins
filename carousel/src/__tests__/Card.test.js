import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Card from "../Card";
import TEST_IMAGES from "../_testCommon.js";
import React from "react";

it("smoke test for card", function () {
  let currCard = TEST_IMAGES[0];
  const { container } = render(
    <Card
      caption={currCard.caption}
      src={currCard.src}
      currNum={1}
      totalNum={1}
    />
  );
  expect(screen.getByText(currCard.caption)).toBeInTheDocument();
});

it("snapshot test for card", function () {
  let currCard = TEST_IMAGES[0];
  const card = renderer
    .create(
      <Card
        caption={currCard.caption}
        src={currCard.src}
        currNum={1}
        totalNum={1}
      />
    )
    .toJSON();
  console.log(card);
  expect(card).toMatchSnapshot();
});
