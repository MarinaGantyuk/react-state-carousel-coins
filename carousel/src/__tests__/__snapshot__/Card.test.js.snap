exports[`snapshot test for card 1`] = `
<div class="Card">
    <h5 class="Card-title">Photo by Richard Pasquarella on Unsplash</h5>
    <img class="Card-image" src="/static/media/image1.08e1519e765278b10548.jpg" alt="Photo by Richard Pasquarella on Unsplash">
    <small class="Card-small">Image 1 of 3.</small>
</div>
`;
